Content.Shared/Movement/Systems/SharedMoverController.cs

```diff
@@ -147,12 +147,22 @@ public override void UpdateAfterSolve(bool prediction, float frameTime)
             LerpRotation(uid, mover, frameTime);
 
             if (!canMove
-                || physicsComponent.BodyStatus != BodyStatus.OnGround
                 || PullableQuery.TryGetComponent(uid, out var pullable) && pullable.BeingPulled)
             {
                 UsedMobMovement[uid] = false;
                 return;
             }
+            
+            if (physicsComponent.BodyStatus != BodyStatus.OnGround)
+            {
+                var ev = new CanInAirMoveEvent(uid);
+                RaiseLocalEvent(uid, ref ev, true);
+                if (!ev.CanMove)
+                {
+                    UsedMobMovement[uid] = false;
+                    return;
+                }
+            }
```

Comment:
This should be using CanMove as that gets cached as well.

Reply:
a bit confused, which CanMove are you referring to here?

Reply:
Raising an event every tick for movement is a really bad idea.

Reply:
Also Mirror ended up doing this for her pr as I didn't see this comment so might be better to just use hers.